committing.rst

```diff
@@ -258,6 +258,41 @@ Developers can apply labels to GitHub pull requests).
 
 .. _cherry_picker.py: https://github.com/python/core-workflow/tree/master/cherry_picker
 
+.. _git_worktrees:
+
+Speeding up development workflow using git worktrees
+----------------------------------------------------
+
+When backporting changes to multiple versions, it's often useful to have more than one checkouts
+of the codebase. This way you can do things like running tests on one version
+while committing a patch on another version. For example: ::
+
+    $ git worktree add ~/cpython_3.6 origin/3.6
+    Preparing /home/user/cpython_3.6 (identifier cpython_3.61)
+    HEAD is now at 49731ef8d2 dict: add iterkeys aliases in 3.5
+
+    $ cd ~/cpython_3.6;
+    $ git cherry-pick 686c146b1ca1eb5c7ef058bb28772ac612a8ec1d
+    [bpo-31620 c48137b107] bpo-31620: have asyncio/queues not leak memory when you've exceptions during waiting
+    Date: Fri Sep 29 01:03:19 2017 +0000
+    files changed, 25 insertions(+)
+    create mode 100644 Misc/NEWS.d/next/Library/2017-10-06-04-35-31.bpo-31620.gksLA1.rst
+
+    $ ./python -m test -j3
+
+    $ git worktree add ~/cpython_2.7 origin/2.7                                 
+    Preparing /home/user/cpython_2.7 (identifier cpython_2.7)
+    HEAD is now at e37fefba8b Revert the last change:
+
+    $ cd ~/cpython_2.7
+
+    # make changes 
+    
+    $ git commit 
```

Comment:
Unless the default highlighting for the devguide is ``shell``, you should specify it.  (You should be able to tell by checking the highlighting of the built docs).

I would remove the empty lines and add more comments: for example what does ``git worktree add ~/cpython_3.6 origin/3.6``?  Something as simple as ``$ # create a new clone of origin/3.6 in the ~/cpython_3.6 dir`` is enough.

A minor nit, but I would prefer if the dir was simply called ``3.6``, IIRC that's what we used in the old devguide and what I've been personally used, since it's shorter and matches the branch name.

Reply:
couldn't really do shell because the lexer couldn't parse all the standard output. lmk if you think we should still do it.